test/unit/ngsiv2/provisioning/device-registration_test.js

```diff
@@ -0,0 +1,344 @@
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/entities?options=upsert')
+                .reply(204);
+
+            iotAgentLib.activate(iotAgentConfig, function(error) {
+                iotAgentLib.clearAll(done);
+            });
+        });
+
+        it('should return all the device\'s information', function(done) {
+            iotAgentLib.register(device1, function(error) {
+                iotAgentLib.getDevice('light1', 'smartGondor', 'gardens', function(error, data) {
+                    should.not.exist(error);
+                    should.exist(data);
+                    data.type.should.equal('Light');
+                    data.name.should.equal('Light:light1');
+                    done();
+                });
+            });
+        });
+    });
+
+    describe('When an unexistent device is requested to the library using its ID', function() {
+        beforeEach(function(done) {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/registrations', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextAvailabilityRequests/registerIoTAgent1.json'))
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            iotAgentLib.activate(iotAgentConfig, function(error) {
+                iotAgentLib.clearAll(done);
+            });
+        });
+
+        it('should return a ENTITY_NOT_FOUND error', function(done) {
+            iotAgentLib.register(device1, function(error) {
+                iotAgentLib.getDevice('lightUnexistent', 'smartGondor', 'gardens', function(error, data) {
+                    should.exist(error);
+                    should.not.exist(data);
+                    error.code.should.equal(404);
+                    error.name.should.equal('DEVICE_NOT_FOUND');
+                    done();
+                });
+            });
+        });
+    });
+
+    describe('When a device is removed from the IoT Agent', function() {
+        beforeEach(function(done) {
+
+            nock.cleanAll();
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .post('/v2/registrations')
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            contextBrokerMock
+                .post('/v2/entities?options=upsert')
+                .reply(204);
+
+            contextBrokerMock
+                .post('/v2/registrations')
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            contextBrokerMock
+                .post('/v2/entities?options=upsert')
```

Comment:
Mock using post without payload. Comment explaining it needed?

Reply:
Fixed with fbc86fa76f4561b711a63053695efd21941cf4e5